src/models/components.d.ts

```diff
@@ -70,20 +71,16 @@ export interface compareTableItem {
   Iron: number;
 }
 
-export interface NutritionTableItem {
+export interface NutritionTableRow {
   nutrient: string;
-  dailyValue: number | null;
-  amount: string | null;
+  dailyValue?: number | null;
+  amount?: number;
+  amountUnit?: string;
+  nutrientGroup: string;
   rdi: {
```

Comment:
I think the entire `rdi` is optional (`?`). If it is present, then `value` and `weight` do exist. Also `weight` can be renamed as `servingUnitSize`

Reply:
udpated